src/rules/use-client.ts

```diff
@@ -208,11 +208,40 @@ const create = Components.detect(
         // @ts-expect-error
         const name = node.object.name;
         const scopeType = context.getScope().type;
-        if (
+
+        // check if the window usage is behind a typeof window === 'undefined' check
+        const conditionalExpressionNode = node.parent?.parent;
+        const isWindowCheck =
+          conditionalExpressionNode?.type === "ConditionalExpression" &&
+          conditionalExpressionNode.test?.type === "BinaryExpression" &&
+          conditionalExpressionNode.test.left?.type === "UnaryExpression" &&
+          conditionalExpressionNode.test.left.operator === "typeof" &&
+          conditionalExpressionNode.test.left.argument?.type === "Identifier" &&
+          conditionalExpressionNode.test.left.argument?.name === "window" &&
+          conditionalExpressionNode.test.right?.type === "Literal" &&
+          conditionalExpressionNode.test.right.value === "undefined";
+
+        // checks to see if it's `typeof window !== 'undefined'` or `typeof window === 'undefined'`
+        const isNegatedWindowCheck =
+          isWindowCheck &&
+          conditionalExpressionNode.test?.type === "BinaryExpression" &&
+          conditionalExpressionNode.test.operator === "!==";
+
+        // checks to see if window is being accessed safely behind a window check
+        const isSafelyBehindWindowCheck =
+          (isWindowCheck &&
+            !isNegatedWindowCheck &&
+            conditionalExpressionNode.alternate === node?.parent) ||
+          (isNegatedWindowCheck &&
+            conditionalExpressionNode.consequent === node?.parent);
```

Comment:
What is `conditionalExpressionNode.consequent === node?.parent` meant to check? Not sure what this is for